File: 2-java/codes/casestudy/emsapp/src/oracle/java/emsapp/entities/Hr.java
package oracle.java.emsapp.entities;

public class Hr extends Employee {

	private double gratuityPayment;

	public Hr() {

	}

	public Hr(String name, int id, double basicPayment, double daPayment, double hraPayment, double gratuityPayment) {
		super(name, id, basicPayment, daPayment, hraPayment);
		this.gratuityPayment = gratuityPayment;
	}

	public double getGratuityPayment() {
		return gratuityPayment;
	}

	public void setGratuityPayment(double gratuityPayment) {
		this.gratuityPayment = gratuityPayment;
	}

	@Override
	public void calculateSalary() {
		super.calculateSalary();
		this.setTotalPayment(gratuityPayment + this.getTotalPayment());
	}
}

File: codes/3-front-end/day-2/payroll-app/src/models/hr.ts
import { Employee } from "./employee";

export default class Hr extends Employee {
    constructor(id: number, name: string, basic: number, da: number, hra: number, public gratuityPayment: number) {
        super(id, name, basic, da, hra)
    }
    calculateSalary() {
        super.calculateSalary()
        this.totalSalary += this.gratuityPayment
    }
}

//export default Hr

//export const a = 100